src/server-request.js

```diff
@@ -0,0 +1,40 @@
+import axios from "axios";
+
+class ApiError extends Error {}
+
+export const serverRequest = async ({
+  url,
+  requestType,
+  dataToOperate,
+  dataToOperateId
+}) => {
+  switch (requestType) {
+    case "GET": {
+      const res = await axios.get(url);
+      if (res.status === 200) {
+        return { response: res };
+      } else {
+        throw new ApiError("Failed to get products");
+      }
+    }
+    case "POST": {
+      const res = await axios.post(url, dataToOperate);
+      if (res.status === 201) {
+        return { response: res };
+      } else {
+        throw new ApiError("Failed to save");
+      }
+    }
+    case "PUT": {
+      const res = await axios.put(`${url}/${dataToOperateId}`, dataToOperate);
+      if (res.status === 200) {
+        return { response: res, errorFlag: false };
+      } else {
+        throw new ApiError("Failed to update");
+      }
+    }
+
+    default:
+      return null;
+  }
+};
```

Comment:
I am confused here the server request is already async then when you call this is again in async fashion. It's redundant no? Am I missing anything?

Reply:
It's an async function, so it will be returning a promise, to wait on the promise to resolve, we have to use await or .then().